packages/client/src/app/challenge/services/challenge.service.js

```diff
@@ -0,0 +1,22 @@
+import { HttpRequestService } from '../../core/index.js';
+
+export class ChallengeService {
+  /** @return {ChallengeService} */
+  static instance() {
+    if (ChallengeService.#instance) return ChallengeService.#instance;
+
+    ChallengeService.#instance = new ChallengeService();
+    return ChallengeService.#instance;
+  }
+
+  /** @type {ChallengeService} */
+  static #instance;
+
+  /**
+   * @param userId {number}
+   * @return {Promise<Array>}
+   */
+  async getChallenges(userId) {
+    return await HttpRequestService.instance().getRequest(`/api/user/${userId}/challenges`);
```

Comment:
We have user in singular and challenges in plural, let's decide on which one to use.